File: docs/competitors/Python/FizzBuzz.py
def fizzbuzz(n: int):

    if n % 3 == 0 and n % 5 == 0:
        return "FizzBuzz"
    elif n % 3 == 0:
        return "Fizz"
    elif n % 5 == 0:
        return "Buzz"
    else:
        return str(n)

def main():
    print("Enter integers:")
    try:
        while True:
            line = input()
            if line.strip() == "":
                continue
            try:
                number = int(line)
                print(fizzbuzz(number))
            except ValueError:
                print("Error: please enter a valid integer.")
    except EOFError:
        print("End of input")

if __name__ == "__main__":
    main()

File: docs/competitors/C++/FizzBuzz.cpp
#include <iostream>
#include <string>
#include <limits>

std::string fizzBuzz(int n)
{
    if (n % 3 == 0 && n % 5 == 0)
    {
        return "FizzBuzz";
    }
    if (n % 3 == 0)
    {
        return "Fizz";
    }
    if (n % 5 == 0)
    {
        return "Buzz";
    }
    return std::to_string(n);
}

bool readInt(int &value)
{
    if (std::cin >> value)
    {
        return true;
    }
    else
    {
        std::cin.clear();
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
        std::cout << "Error: please enter a valid integer." << std::endl;
        return false;
    }
}

int main()
{
    std::cout << "Enter integers:" << std::endl;
    int number;
    while (true)
    {
        if (readInt(number))
        {
            std::cout << fizzBuzz(number) << std::endl;
        }
        else if (std::cin.eof())
        {
            break;
        }
    }

    return 0;
}
